File: 2026-01/src/day_30/prms_118667/Solution.java
package day_30.prms_118667;

import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        Queue<Integer> q1 = arrToQueue(queue1);
        Queue<Integer> q2 = arrToQueue(queue2);

        long sum1 = addition(queue1);
        long sum2 = addition(queue2);

        int cnt = 0;
        int max = (q1.size() + q2.size()) * 2;

        while (cnt < max) {
            if (sum1 < sum2) {
                sum1 += q2.peek();
                sum2 -= q2.peek();
                q1.offer(q2.poll());
            } else if (sum1 > sum2) {
                sum1 -= q1.peek();
                sum2 += q1.peek();
                q2.offer(q1.poll());
            } else {
                return cnt;
            }

            cnt++;
        }

        return -1;
    }

    static Queue<Integer> arrToQueue(int[] arr) {
        Queue<Integer> q = new ArrayDeque<>();

        for (int n : arr) {
            q.offer(n);
        }

        return q;
    }

    static long addition(int[] arr) {
        long sum = 0;

        for (int n : arr) {
            sum += n;
        }

        return sum;
    }
}

File: 2026-01/src/day_30/prms_118667/Solution.cpp
#include <queue>
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    queue<int> q1;
    queue<int> q2;
    for (int x : queue1) q1.push(x);
    for (int x : queue2) q2.push(x);

    long long sum1 = 0;
    long long sum2 = 0;
    for (int x : queue1) sum1 += x;
    for (int x : queue2) sum2 += x;

    int cnt = 0;
    int mx = (q1.size() + q2.size()) * 2;

    while (cnt < mx) {
        if (sum1 < sum2) {
            sum1 += q2.front();
            sum2 -= q2.front();
            q1.push(q2.front());
            q2.pop();
        } else if (sum1 > sum2) {
            sum1 -= q1.front();
            sum2 += q1.front();
            q2.push(q1.front());
            q1.pop();
        } else {
            return cnt;
        }

        cnt++;
    }

    return -1;
}
